classes/Hook.php

```diff
@@ -126,10 +126,20 @@ class HookCore extends ObjectModel
     public function add($autodate = true, $null_values = false)
     {
         Cache::clean('hook_idsbyname');
+        Cache::clean('hook_idsbyname_withalias');
+        Cache::clean('active_hooks');
 
         return parent::add($autodate, $null_values);
     }
 
+    public function clearCache($all = false)
+    {
+        Cache::clean('hook_idsbyname');
```

Comment:
```suggestion
        Cache::clean('hook_*');
```

Reply:
Just for information, but you can clear multiple cache keys using the wildcard `*` Not mandatory though

Reply:
@jolelievre
That is the change that @sowbiba requested https://github.com/PrestaShop/PrestaShop/pull/26748#discussion_r770592862. He doesn't like the wildcard `*`

Reply:
Don't mind. It's just cache, supposed to be easy to rebuild. You can revert the change

Reply:
If you use the following wording you don't have to implement the clearCache method
```suggestion
        Cache::clean('objectmodel_hook_idsbyname');
```
It will rely only on this object model and it will be easier to clear, it's also more consistent because you've less chance for someone to already use this cache keys

Reply:
You're also able to use const in order to don't copy & paste strings and make sure the wording is the same everywhere

Reply:
@lmeyer1  Hi, i don't know where to post this (not realy usual with github), but since the update of this file, I have a blog module that ends on a 404. I put back the Hook.php file from version 1.7.8.3 and no more probleme. Have you any idea to fix this issue ?

Reply:
@Djolhan1 Hello, please open a new issue, thank you https://github.com/PrestaShop/PrestaShop/issues

Reply:
> @Djolhan1 Hello, please open a new issue, thank you https://github.com/PrestaShop/PrestaShop/issues

Thank for your answer, Done, hope it help : https://github.com/PrestaShop/PrestaShop/issues/27872